packages/engine/crawlers/sui/collectibles.ts

```diff
@@ -0,0 +1,151 @@
+import type {
+	JsonRpcProvider,
+	PaginatedObjectsResponse,
+	SuiObjectData,
+} from '@mysten/sui.js';
+import { Networks } from '@walless/core';
+import type { CollectibleDocument } from '@walless/store';
+
+export const getSuiCollectibles = async (
+	connection: JsonRpcProvider,
+	owner: string,
+) => {
+	const collectibleDocuments: CollectibleDocument[] = [];
+	let objectResponses: PaginatedObjectsResponse['data'] = [];
+	let nextCursor: PaginatedObjectsResponse['nextCursor'] | undefined;
+	let cursor: PaginatedObjectsResponse['nextCursor'] | undefined;
+	while (nextCursor !== null) {
+		const allOwnedObjects = await connection.getOwnedObjects({
+			owner,
+			cursor,
+			options: {
+				showType: true,
+				showOwner: true,
+				showDisplay: true,
+				showContent: true,
+			},
+		});
+
+		cursor = allOwnedObjects.nextCursor;
+
+		if (allOwnedObjects.hasNextPage && nextCursor !== cursor) {
+			nextCursor = cursor;
+		} else {
+			nextCursor = null;
+		}
+
+		objectResponses = [...objectResponses, ...allOwnedObjects.data];
```

Comment:
In this part of code, the main logic is `objectResponses = [...objectResponses, ...allOwnedObjects.data];` should be placed on top of remaining which is checking next cursor.

Also, does it really need `cursor` variable, I think just assign nextCusor continuously, if it does not have next, it will be null or undefined

```
		cursor = allOwnedObjects.nextCursor;

		if (allOwnedObjects.hasNextPage && nextCursor !== cursor) {
			nextCursor = cursor;
		} else {
			nextCursor = null;
		}
``` 
to
```
nextCursor = allOwnedObjects.nextCursor;
```